### PR TITLE
Generate bindings of Image::to_rgb8/to_rgba8/to_rgba8_premultiplied f…

### LINKED ISSUE #6399: Generate bindings of `Image::to_rgb8/to_rgba8/to_rgba8_premultiplied` for C++
They are missing in C++ side.

### PR DESCRIPTION
**Pull request by tronical**
…or C++

ChangeLog: [C++] Add `Image::to_rgb8/to_rgba8/to_rgba8_premultiplied` pixel buffer accessors.

Fixes #6399

### CODE REVIEW

File: internal/core/graphics/image.rs [hunk 1]
```diff
@@ -1324,6 +1324,51 @@ pub(crate) mod ffi {
     ) {
         image.set_nine_slice_edges(top, right, bottom, left);
     }
+
+    #[no_mangle]
+    pub unsafe extern "C" fn slint_image_to_rgb8(
```
Comment (ogoffart) [inline internal/core/graphics/image.rs:1329]: technically, these function don't need to be unsafe. They don't have invariant that is not expressed in the types.

**Review by ogoffart — APPROVED** [review]

File: internal/core/graphics/image.rs (hunk 1 above)
Comment (tronical) [reply internal/core/graphics/image.rs:1329]: Ahh, copy & paste :). I'll remove it.

File: internal/core/graphics/image.rs [hunk 2]
```diff
@@ -1324,6 +1324,51 @@ pub(crate) mod ffi {
...
     }
+
+    #[no_mangle]
+    pub extern "C" fn slint_image_to_rgb8(
+        image: &Image,
+        data: &mut SharedVector<Rgb8Pixel>,
+        width: &mut u32,
+        height: &mut u32,
+    ) -> bool {
+        image.to_rgb8().map_or(false, |pixel_buffer| {
```
Comment (ogoffart) [inline internal/core/graphics/image.rs:1335]: this is `is_some_and()`
But it doesn't have to be changed now.  @hunger will change it later when he applies clippy suggestion :-)

**Review by ogoffart — APPROVED** [review]

File: internal/core/graphics/image.rs (hunk 2 above)
Comment (tronical) [reply internal/core/graphics/image.rs:1335]: Hehe ok :)

**Pull request merged**